freud/density.pyx

```diff
@@ -228,17 +228,33 @@ cdef class GaussianDensity(_Compute):
         """:class:`freud.box.Box`: Box used in the calculation."""
         return freud.box.BoxFromCPP(self.thisptr.getBox())
 
-    def compute(self, system):
+    def compute(self, system, values=None):
         R"""Calculates the Gaussian blur for the specified points.
 
         Args:
             system:
                 Any object that is a valid argument to
                 :class:`freud.locality.NeighborQuery.from_system`.
+            values ((:math:`N_{points}`) :class:`numpy.ndarray`):
+                Values associated with the system points used to calculate the
+                convolution. Calculates Gaussian blur if :code:`None`.
+                (Default value = :code:`None`).
         """
         cdef freud.locality.NeighborQuery nq = \
             freud.locality.NeighborQuery.from_system(system)
-        self.thisptr.compute(nq.get_ptr())
+
+        if values is None:
+            values = freud.util._convert_array(
+                np.ones(nq.points.shape[0]), shape=(nq.points.shape[0], ),
+                dtype=np.float32)
```

Comment:
1. We can avoid an unnecessary double-float conversion here. We should specify the dtype for `np.ones` since it defaults to `float64`, but we don't need to specify the dtype for `_convert_array` since it defaults to `float32`. 
2. We don't need to check the shape in `_convert_array` since we already know its shape is correct (by construction). We only have to check the shape of a user-provided array of values.

```suggestion
                np.ones(nq.points.shape[0], dtype=np.float32))
```